src/main/kotlin/net/atos/zac/app/zaken/converter/historie/RESTZaakHistoriePartialUpdateConverter.kt

```diff
@@ -5,42 +5,48 @@ import net.atos.client.vrl.VrlClientService
 import net.atos.client.vrl.model.generated.CommunicatieKanaal
 import net.atos.client.zgw.shared.model.audit.ZRCAuditTrailRegel
 import net.atos.client.zgw.shared.util.URIUtil
+import net.atos.client.zgw.zrc.ZRCClientService
 import net.atos.client.zgw.zrc.model.Geometry
 import net.atos.zac.app.audit.model.RESTHistorieActie
 import net.atos.zac.app.audit.model.RESTHistorieRegel
 import java.net.URI
 
 private const val COMMUNICATIEKANAAL = "communicatiekanaal"
 private const val ZAAKGEOMETRIE = "zaakgeometrie"
+private const val HOOFDZAAK = "hoofdzaak"
+private const val RELEVANTE_ANDERE_ZAKEN = "relevanteAndereZaken"
 
 class RESTZaakHistoriePartialUpdateConverter @Inject constructor(
-    private val vrlClientService: VrlClientService
+    private val vrlClientService: VrlClientService,
+    private val zrcClientService: ZRCClientService
 ) {
     fun convertPartialUpdate(
         auditTrail: ZRCAuditTrailRegel,
         actie: RESTHistorieActie?,
         old: Map<*, *>,
         new: Map<*, *>
     ) =
-        old.diff(new).map { convertLine(auditTrail, actie, it) }
+        old.diff(new).mapNotNull { convertLine(auditTrail, actie, it) }
 
     private fun convertLine(
         auditTrail: ZRCAuditTrailRegel,
         actie: RESTHistorieActie?,
-        change: Map.Entry<Any?, Pair<*, Any?>>
-    ): RESTHistorieRegel =
-        RESTHistorieRegel(
-            change.key.toString(),
-            convertValue(change.key, change.value.first),
-            convertValue(change.key, change.value.second)
-        ).apply {
-            datumTijd = auditTrail.aanmaakdatum
-            door = auditTrail.gebruikersWeergave
-            toelichting = auditTrail.toelichting
-            this.actie = actie
+        change: Map.Entry<Any?, Pair<Any?, Any?>>
+    ): RESTHistorieRegel? =
```

Comment:
So this can return null? seems a bit weird to me?

Reply:
We want to filter out the zaakobject for the Product aanvraag. We do that by returning null. I can also make this function return a list of either one or zero if you think that's better

Reply:
ah right. let's leave it like this. I think this is fine. maybe add a little javadoc indicating that it returns null in that case?